fastn-ds/src/lib.rs

```diff
@@ -359,6 +364,141 @@ impl DocumentStore {
         )
         .await)
     }
+
+    #[tracing::instrument(skip_all)]
+    pub async fn http_get_with_cookie(
+        &self,
+        url: &str,
+        cookie: Option<String>,
+        headers: &std::collections::HashMap<String, String>,
+        use_cache: bool,
+    ) -> fastn_ds::DSResult<(fastn_ds::DSResult<Vec<u8>>, Vec<String>)> {
+        let mut cookies = vec![];
+
+        if use_cache && fastn_ds::http::NOT_FOUND_CACHE.read().contains(url) {
+            return Ok((
+                Err(fastn_ds::DSError::GenericError {
+                    message: "page not found, cached".to_string(),
+                }),
+                cookies,
+            ));
+        }
+
+        tracing::info!(url = url);
+        let mut req_headers = reqwest::header::HeaderMap::new();
+        req_headers.insert(
+            reqwest::header::USER_AGENT,
+            reqwest::header::HeaderValue::from_static("fastn"),
+        );
+        if let Some(cookie) = cookie {
+            req_headers.insert(
+                reqwest::header::COOKIE,
+                reqwest::header::HeaderValue::from_str(cookie.as_str()).unwrap(),
+            );
+        }
+
+        for (key, value) in headers.iter() {
+            req_headers.insert(
+                reqwest::header::HeaderName::from_bytes(key.as_bytes()).unwrap(),
+                reqwest::header::HeaderValue::from_str(value.as_str()).unwrap(),
+            );
+        }
+
+        let c = reqwest::Client::builder()
+            .default_headers(req_headers)
+            .build()?;
+
+        // if we are not able to send the request, we could not have read the cookie
+        let res = c.get(url).send().await?;
+
+        res.headers().iter().for_each(|(k, v)| {
+            if k.as_str().eq("set-cookie") {
+                if let Ok(v) = v.to_str() {
+                    cookies.push(v.to_string());
+                }
+            }
+        });
+
+        if !res.status().eq(&reqwest::StatusCode::OK) {
+            let message = format!(
+                "url: {}, response_status: {}, response: {:?}",
+                url,
+                res.status(),
+                res.text().await
+            );
+            tracing::error!(url = url, msg = message);
+
+            if use_cache {
+                fastn_ds::http::NOT_FOUND_CACHE
+                    .write()
+                    .insert(url.to_string());
+            }
+
+            return Ok((Err(fastn_ds::DSError::GenericError { message }), cookies));
+        }
+        tracing::info!(msg = "returning success", url = url);
+        Ok((Ok(res.bytes().await?.into()), cookies))
+    }
+
+    #[tracing::instrument(skip_all)]
+    pub async fn http_post_with_cookie(
```

Comment:
Can you make them delegate these functions to the already created http method so I just have to overwrite http and not many functions?